src/lib/all_type_variant.hpp

```diff
@@ -32,6 +32,7 @@ namespace detail {
 
 // clang-format off
 #define DATA_TYPE_INFO                  \
+  ((uint8_t,     Bool,       "bool"))   \
```

Comment:
This is not a good idea since this adds `bool` as a possible data type to all our column templates.

Reply:
If you need this for your JIT variant, use the `DATA_TYPES` macro or even better the `data_types` tuple and append `bool` and use that to create your JIT variant.

Reply:
@mjendruk I did that before and @mrks suggested to merge both datatype definitions.
I don't really have a strong opinion about this. What are the downsides of adding it here? More template instantiations --> slower compile times or sth else? It does not seem to break any tests...

Reply:
Yes, every template that we instantiate for our data types gets an additional instantiation. This does not make sense, because it adds a lot of compile time for nothing.

Reply:
Hmm, I didn't think about the compile time. Just thought that having a "bool" type doesn't hurt. But if it also causes problems with the tests, maybe we should stick to the original version. :/

Reply:
What about removing it from the `DATA_TYPE_INFO`, but keeping it in the enum? I know, that is kind of inconsistent, but way we can at least get rid of all the `jit_data_type_to_data_type` conversions.

Reply:
I could live with that if we add a comment on why we did it.

Reply:
I don’t know. Thoughtfully implemented, `jit_data_type_to_data_type` could be a simple static_cast.

Reply:
That feels pretty (subjectively) hacky to me, but I am happy to do it either way.

Reply:
Does it? If you create `DataType` and `JitType` using the same macro (`DATA_TYPES`) with the only difference that you add add another type to the list for `JitType`, there is no reason it could break at some point.

Reply:
But what happens if you statically cast the enum member, that does exist in JitType, but not DataType? With the `std::unordered_map` we would at least get a runtime exception, whether I think a static_cast will just lead to undefined behavior?

Reply:
```c++
DataType to_data_type(const JitType& jit_type) {
  Assert(jit_type != JitType::Bool, "");  // or ...
  Assert(static_cast<uint8_t>(jit_type) <= static_cast<uint8_t>(JitType::FirstAdditionalType), "");

  return static_cast<DataType>(jit_type);
}
```